Headers/GNUstepBase/GSBlocks.h

```diff
@@ -100,13 +100,23 @@ typedef retTy(^name)()
   } *name
 
 
-#define CALL_BLOCK(block, args...) block->invoke(block, args)
-#define CALL_BLOCK_NO_ARGS(block) block->invoke(block)
+#define CALL_NON_NULL_BLOCK(block, args...) block->invoke(block, args)
+#define CALL_NON_NULL_BLOCK_NO_ARGS(block) block->invoke(block)
 #define BLOCK_SCOPE
 #endif /* GCC_VERSION >= 3000 */
 
 #endif /* __has_feature(blocks) */
 
+#define CALL_BLOCK(block, args...) do { \
+  if (NULL != block) \
+    CALL_NON_NULL_BLOCK(block, args); \
```

Comment:
You just need a "return", plus an else case that will return "nil". This is not perfect, but better than this.

Reply:
do you suggest that I change it to `#define CALL_BLOCK(block, args...) ((NULL != block) ? CALL_NON_NULL_BLOCK(block, args) : nil);` instead ?

Reply:
That looks good to me; clear and concise.

Reply:
updated

Reply:
@rfm  can this be merged?